profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+<?php
+
+/**
+ * @file
+ * Code for the NewsroomHelper class.
+ */
+
+/**
+ * NewsroomHelper class.
+ */
+class NewsroomHelper {
+
+  /**
+   * Merges a term_tid and its children into one array.
+   *
+   * @param int $term_tid
+   *   Term id.
+   * @param int $term_vid
+   *   Vid id.
+   *
+   * @return array.
+   *   tids
+   */
+  public static function getTermChildren($term_tid, $term_vid = NULL) {
+    // We need parent ID also in the output array to use it in query.
+    $parent_children_tids = array($term_tid);
+    $children_tids = array_keys(taxonomy_get_children($term_tid, $term_vid));
+    if (count($children_tids) > 0) {
+      $parent_children_tids = array_merge($parent_children_tids, $children_tids);
+    }
+
+    return $parent_children_tids;
+  }
+
+  /**
+   * Helper function to serialize PDO object into an array.
+   */
+  public static function convertPdoToArray($query) {
+    $data = array();
+    if ($query) {
+      while ($result = $query->fetchAssoc()) {
+        $data[] = $result;
+      }
+    }
+
+    return $data;
+  }
+
+  /**
+   * Returns newsroom URL.
+   *
+   * @return string
+   *   Newsroom URL.
+   */
+  public static function getNewsroomUrl() {
+    return 'newsroom/' . implode('/', func_get_args());
```

Comment:
Because it is important part of the feature and we are sure it is there. It is the same case as node/[nid]